test/e2e_node/runner/run_e2e.go

```diff
@@ -0,0 +1,125 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/test/e2e_node/gcloud"
+)
+
+type RunFunc func(host string, port string) ([]byte, error)
+type Result struct {
+	host   string
+	output []byte
+	err    error
+}
+
+var u = sync.WaitGroup{}
+var zone = flag.String("zone", "", "gce zone the hosts live in")
+var hosts = flag.String("hosts", "", "hosts to test")
+var dir = flag.String("dir", "../", "test directory")
+var wait = flag.Bool("wait", false, "if true, wait for input before running tests")
+
+func main() {
+	flag.Parse()
+	if *hosts == "" {
+		glog.Fatalf("Must specific --hosts flag")
+	}
+	if *zone == "" {
+		glog.Fatalf("Must specific --zone flag")
+	}
+
+	// Build the go code
+	out, err := exec.Command("../../../hack/build-go.sh").CombinedOutput()
+	if err != nil {
+		glog.Fatalf("Failed to build go packages %s: %v", out, err)
+	}
+
+	// Copy kubelet to each host and run test
+	if *wait {
+		u.Add(1)
+	}
+
+	w := sync.WaitGroup{}
+	for _, h := range strings.Split(*hosts, ",") {
+		w.Add(1)
+		go func(host string) {
+			out, err := runTests(host)
+			glog.Infof("Finished Test Suite %s %v", out, err)
+			w.Done()
+		}(h)
+	}
+
+	// Maybe wait for user input before running tests
+	if *wait {
+		WaitForUser()
+	}
+
+	// Wait for the tests to finish
+	w.Wait()
+	glog.Infof("All hosts finished")
+}
+
+func WaitForUser() {
+	scanner := bufio.NewScanner(os.Stdin)
+	fmt.Printf("Enter \"Done\" to run tests\n")
```

Comment:
No good reason, to be non-standard :P I can update.  I am also open to deleting this functionality if it complicates things.